sdk/opentelemetry-sdk.gemspec

```diff
@@ -25,6 +25,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ['lib']
   spec.required_ruby_version = '>= 2.5.0'
 
+  spec.add_dependency 'google-cloud-env'
```

Comment:
This is the first 3rd party runtime dependency we're adding. I'm not saying we shouldn't add it, just pointing that out.

Reply:
I actually had a conversation with @fbogsany regarding this after I pushed this up, which is why I converted it back to draft.  He advised me that this functionality should be extracted out of the SDK and made into its own gem so not to add dependencies for something that will be optional.

My plan here will be to create a `detectors/` folder at the root of the repository containing the gem and it will include the auto detection functionality as well as all the detectors.  

What do you think @mwear ?

Reply:
I think that makes sense and generally mirrors what we're doing in JS where resources are their own npm package.

Reply:
The work to split it out into a separate gem has been pushed up now.